ui/src/components/chat/compose/chatbar.rs

```diff
@@ -528,10 +534,23 @@ pub fn get_chatbar<'a>(cx: &'a Scoped<'a, super::ComposeProps>) -> Element<'a> {
                                 let mut new_files_to_upload: Vec<_> = state.read().get_active_chat().map(|f| f.files_attached_to_send)
                                     .unwrap_or_default()
                                     .iter()
-                                    .filter(|file_name| !new_files.contains(file_name))
+                                    .filter(|file_location| {
```

Comment:
not a blocker but if you `map` `new_files` to a `Location::Disk` first, using `filter` on the files retrieved from `state` results in a much smaller closure. The naming is also better. Also using `drain(..)` removes the need to `clone` the files being iterated over. 

```
if let Some(new_files) = FileDialog::new()
    .set_directory(dirs::home_dir().unwrap_or_default())
    .pick_files()
{
    let new_files: Vec<Location> = new_files.iter()
    .map(|path| Location::Disk { path: path.clone() })
    .collect();

    let mut current_files: Vec<_> =  state.read().get_active_chat().map(|f| f.files_attached_to_send)
    .unwrap_or_default().drain(..).filter(|x| !new_files.contains(x)).collect();
    current_files.extend(new_files);
    state.write().mutate(Action::SetChatAttachments(chat_id, current_files));
    update_send();
}
```